File: packages/transform/src/features/http/HttpHeadersTransformer.ts
import { HttpHeadersProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpHeadersTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "http.headers",
      write: HttpHeadersProgrammer.write,
    });
}

File: packages/transform/src/features/http/HttpHeadersTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var HttpHeadersTransformer = httpHeadersTransformerNamespace{}

type httpHeadersTransformerNamespace struct{}

func (httpHeadersTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpHeadersProgrammer.Write(call.TypeText)
}
